google/cloud/storage/internal/hash_function_impl.cc

```diff
@@ -152,6 +158,39 @@ Status MD5HashFunction::Update(std::int64_t offset, absl::Cord const& buffer,
   return InvalidArgumentError("mismatched offset", GCP_ERROR_INFO());
 }
 
+#ifdef _WIN32
```

Comment:
I think we should split the MD5 implementation to separate files for OpenSSL and CGN / bcrypt too. Here and in `storage/internal/openssl_util.cc`.

Reply:
The MD5 code in the second case is tiny. I don't think it is worth splitting it into two files.

Reply:
Split `hash_function_impl.cc`.